src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,382 @@
+                cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<Solution> ChangeSolutionAsync(
+            Document document,
+            TInvocationSyntax calleeMethodInvocationNode,
+            TMethodDeclarationSyntax calleeMethodNode,
+            InlineMethodContext inlineMethodContext,
+            bool removeCalleeDeclarationNode,
+            CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var solution = document.Project.Solution;
+            var solutionEditor = new SolutionEditor(solution);
+            var documentEditor = await solutionEditor.GetDocumentEditorAsync(document.Id, cancellationToken).ConfigureAwait(false);
+            var statementContainsCalleeInvocationExpression = inlineMethodContext.StatementContainingCallee;
+            foreach (var statement in inlineMethodContext.StatementsToInsertBeforeCallee)
+            {
+                documentEditor.InsertBefore(
+                    statementContainsCalleeInvocationExpression,
+                    statement);
+            }
+
+            var syntaxNodeToReplace = inlineMethodContext.SyntaxNodeToReplace;
+            var inlineSyntaxNode = inlineMethodContext.InlineSyntaxNode;
+            documentEditor.ReplaceNode(syntaxNodeToReplace, inlineSyntaxNode);
+
+            // If the inline content has 'await' expression, then make sure the caller is change to 'async' method
+            // if its return type is awaitable. In all other cases, do nothing.
+            if (inlineMethodContext.ContainsAwaitExpression)
+            {
+                var callerMethodSymbol = GetContainingMethodSymbol(calleeMethodInvocationNode, semanticModel, cancellationToken);
+                if (callerMethodSymbol?.IsAsync == false)
+                {
+                    var symbolDeclarationService = document.GetRequiredLanguageService<ISymbolDeclarationService>();
+                    var callerMethodReferences = symbolDeclarationService.GetDeclarations(callerMethodSymbol);
+                    if (callerMethodReferences.Length == 1)
+                    {
+                        var callerMethodNode = await callerMethodReferences[0].GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+                        if (callerMethodSymbol.ReturnsVoid || callerMethodSymbol.IsAwaitableNonDynamic(semanticModel, callerMethodNode.SpanStart))
+                        {
+                            var declarationModifiers = DeclarationModifiers.From(callerMethodSymbol).WithAsync(true);
+                            documentEditor.SetModifiers(callerMethodNode, declarationModifiers);
+                        }
+                    }
+                }
+            }
+
+            if (removeCalleeDeclarationNode)
+            {
+                var documentId = solution.GetDocumentId(calleeMethodNode.SyntaxTree);
```

Comment:
do you have tests fort eh multi-doc case?

Reply:
Right now no, I will add that